File: 2022/day01_1.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any


def solution(content: list[str]) -> Any:
    elves: list[list[int]] = [[]]
    for line in content:
        if line.strip() == "":
            elves += [[]]
        else:
            elves[-1] += [int(line)]
    return sorted([sum(elf) for elf in elves], reverse=True)[0]


def main() -> None:
    with open("day01.dat", "r", encoding="utf8") as file:
        print(solution(file.readlines()))


if __name__ == "__main__":
    main()

File: 2022/day01-1.ts
function solution(content: string) {
  return content
    .split("\n")
    .reduce(
      (acc, cur) => {
        if (cur.trim() === "") {
          return [...acc, []];
        } else {
          acc.at(-1)?.push(parseInt(cur));
          return [...acc];
        }
      },
      [[]] as number[][]
    )
    .map((elf) => elf.reduce((cur, acc) => cur + acc, 0))
    .toSorted((a, b) => b - a)
    .at(0);
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
